docs/src/proposals/comprehensive-compute-fees.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Comprehensive Compute Fees
+---
+
+## Motivation
+
+The current fee structure lacks a comprehensive account of the work required by
+a validator to process a transaction.  The fee structure is only based on the
+number of signatures in a transaction but is meant to account for the work that
+the validator must perform to validate each transaction.  The validator performs
+a lot more user-defined work than just signature verification.  The work to
+process a transaction typically includes signature verifications, account
+locking, account loading, and instruction processing.
+
+## Proposed Solution
+
+### New fee structure
+
+In addition to signature checking, the other work required to process an
+instruction should be taken into account.  To do this, the total fee should be
+an accumulated cost of all the work.  Each piece of work can be
+measured/accounted for in a way that makes sense for the type of work involved.
+
+A new fee structure could include:
+1. flat fee per signature
+2. flat fee for each write lock
+3. per-byte fee for the amount of data in each loaded account
+4. per-compute-unit fee for the number of compute units used to process an
+   instruction
+
```

Comment:
There's one more here: there are some instructions that require an extra fee, currently just the [secp256k1_program](https://github.com/solana-labs/solana/blob/7cea2c446656510da7cbe943d3e87c18e794a76e/sdk/program/src/fee_calculator.rs#L24).  We consider that to be an extra signature fee right now but generalizing into the concept of there exists programs that cost X to use them seem worthwhile if we're redoing the fee structure